src/fdc/MSXtar.cc

```diff
@@ -218,10 +247,17 @@ FatCluster MSXtar::readFAT(Cluster cluster) const
 	std::span<const uint8_t> data{fatBuffer[0].raw.data(), sectorsPerFat * size_t(SECTOR_SIZE)};
 
 	unsigned index = FAT::FIRST_CLUSTER + cluster.index;
-	auto p = subspan<2>(data, (index * 3) / 2);
-	unsigned value = (index & 1)
-	                ? (p[0] >> 4) + (p[1] << 4)
-	                : p[0] + ((p[1] & 0x0F) << 8);
+	unsigned value;
+
+	if (fat16) {
+		auto p = subspan<2>(data, index * 2);
+		value = p[0] | p[1] << 8;
+	} else {
+		auto p = subspan<2>(data, (index * 3) / 2);
+		value = (index & 1)
+		      ? (p[0] >> 4) + (p[1] << 4)
+		      : p[0] + ((p[1] & 0x0F) << 8);
+	}
```

Comment:
Very much a detail. I personally try to avoid having (even temporary) uninitialized variables. Here it can be avoided with a construct like:
```
unsigned value = [&] {
    if (fat16) {
        ...
        return p[0] | p[1] << 8;
    } else {
        ...
        return ...;
    }
}();
```
In other words use an "immediately invoked function expression (IIFE)". But I understand this can make the code less readable if you're not familiar with the idiom.

Bonus: that would allow to make `value` `const`. Personally I don't have the habit of making local variables `const` whenever possible (this rule has both proponents and opponents).